### ISSUE TITLE
iD incorrectly offers to upgrade London Underground stations to London Overground

### ISSUE THREAD

**Issue by peternewman** [osmlab/name-suggestion-index#10167]
https://nsi.guide/index.html?t=transit&k=public_transport&v=station_subway&tt=london%20under#londonunderground-05dd5a
to
https://nsi.guide/index.html?t=transit&k=railway&v=station&tt=london%20overg#londonoverground-4a7f23

For at least Oxford Circus:
https://www.openstreetmap.org/node/5614077964

Although not Great Portland Street (presumably due to the lack of operator):
https://www.openstreetmap.org/node/6414039764

I assume this is because an underground station:
```
public_transport=station
railway=station
station=subway
subway=yes
```

Is essentially a subset of a railway station:
```
public_transport=station
railway=station
train=yes
```

I'm not sure how to fix it short of tagging `train=no` it turns out you can't actually solve it with `not:operator:wikidata` as it also matches the Elizabeth Line ( https://nsi.guide/index.html?t=transit&k=railway&v=station&tt=eliz#elizabethline-abdde3 ) and it doesn't support semicolon or comma listing of multiple values.

**Comment by Snowysauce** [comment]
This is probably related to the iD-specific behavior described at https://github.com/osmlab/name-suggestion-index/issues/9996#issuecomment-2424196068, so I'm not sure if anything can be done about this on NSI's end.

**Comment by UKChris-osm** [comment]
Great Portland Street doesn't include a `network:wikidata` at the moment, so I expect that's why no suggestions are taking place. Adding the underground `network:wikidata` causes the same suggestions to override with the Overground and Elizabeth Line.

Clicking `not` just seems to rotate between Elizabeth and Overground continuously.

**Comment by bhousel** [comment]
Yeah this does seem to be an issue only with iD.  If the station already has network and operator values, I don't know why it would want to switch London Underground and London Overground.

I tested this station in Rapid and I don't get the issue:
https://rapideditor.org/canary#map=20.24/51.51544/-0.14196&background=Bing&datasets=fbRoads,msBuildings&disable_features=boundaries&id=n5614077964

**Comment by matkoniecz** [comment]
> For at least Oxford Circus:
> https://www.openstreetmap.org/node/5614077964

this one seems fixed based on my testing

**Comment by matkoniecz** [comment]
> Yeah this does seem to be an issue only with iD. 

is there is issue reported in iD repository? I see one in id-tagging-schema but I have no idea how any change to iD tagging schema would fix this bug

**Comment by matkoniecz** [comment]
>  https://www.openstreetmap.org/node/5614077964

it now wants to apply `network=Elizabeth line`

is it also wrong?

BTW, https://github.com/osmlab/name-suggestion-index/issues/10469#issuecomment-3678706258 asked for live examples - request is from someone trying to fix an iD bug

**Comment by peternewman** [comment]
> > https://www.openstreetmap.org/node/5614077964
> 
> it now wants to apply `network=Elizabeth line`
> 
> is it also wrong?

Yes. Transport for London has three to five "railway" networks (depending on how you count them):

- London Underground
- London Overground
- ~Wombling free~
- Elizabeth Line
- Docklands Light Railway (DLR) - possibly part of the Underground
- Trams

I think it's just cycling because they all match some bits, and because multiple not tags aren't supported, you can't avoid it that way. I think the Elizabeth Line is more recently added as a network to NSI.

**Comment by matkoniecz** [comment]
EDIT: https://github.com/openstreetmap/iD/issues/11649 exists already

=============================

So lets take look at https://www.openstreetmap.org/node/5614077964 which is part of London Underground (Q20075) and is not on Elizabeth line (Q111297173)

iD pushes it to be retagged as in Elizabeth line network, why?


https://github.com/osmlab/name-suggestion-index/blob/c1ea719828390437b4a4f8838e7d0efcbefaf82e/data/transit/route/subway.json#L254-L262

https://github.com/osmlab/name-suggestion-index/blob/c1ea719828390437b4a4f8838e7d0efcbefaf82e/data/transit/route/train.json#L582-L591

so it seems that iD bug is that despite

```
   "network:wikidata": "Q20075", 
   "operator": "Transport for London", 
   "operator:wikidata": "Q682520", 
```

being tagged at https://www.openstreetmap.org/node/5614077964/history it wants to change it to

```
   "network:wikidata": "Q111297173", 
   "operator": "Transport for London", 
   "operator:wikidata": "Q682520", 
```

with no (obvious) good reason

**Comment by UKChris-osm** [comment]
Could it be simply because of alphabetical order? Both the `Elizabeth Line` and the `London Overground` are assigned `TFL` as the `operator`, as well as the `London Underground`, so if ID is matching up `TFL` as the operator as the priority match, then the `Elizabeth Line` is simply first in line alphabetically?

So in the above example for [Oxford Circus](https://www.openstreetmap.org/node/5614077964), the matcher is saying "oh, this is TFL, and TFL have three possible options, and the Elizabeth Line is first in the list"?

<img width="1136" height="559" alt="Image" src="https://github.com/user-attachments/assets/0a89f80e-4e84-4bd7-befc-3af4037a0f99" />

**Comment by matkoniecz** [comment]
@peternewman looking into it, this may be simply OSM data issue. See https://www.openstreetmap.org/note/5536750

(or maybe it is not, but before investigating it deeper it should be fixed to make situation less confused)
